### DISCUSSION TITLE
Need help understanding model inputs

### DISCUSSION THREAD

**Discussion by nmarquesin** [NIVANorge/lake_liming_app#15] [General]
I need help to understand this about our model: if the goal is to figure out how much of a given lime product needs to be added to a lake to reach a desired pH, then why are lime dose and liming frequency (overdosing) model inputs instead of outputs? Would it be possible to have desired min pH in x months or desired pH range to be user inputs instead?

Our current model plots different residency times, does that mean that this part of our scope has changed? "(...) the application will by default assume wet spreading of lime by boat to a lake with residence time 0.7 years. "

**Comment by JamesSample** [comment]
Hi @nmarquesin,

Good questions!

> if the goal is to figure out how much of a given lime product needs to be added to a lake to reach a desired pH, then why are lime dose and liming frequency (overdosing) model inputs instead of outputs? Would it be possible to have desired min pH in x months or desired pH range to be user inputs instead?

I think you're correct that this is the ultimate goal, but there are lots of different ways this model can be used and the output presented (see e.g. the PDF screenshot [here](https://github.com/NIVANorge/lake_liming_app/issues/7)). "Kalkoptimalisering" is one of the features of the old TPKALK model, but it's not the main component MDir have asked us to focus on initially. Regardless, all options depend on us being able to take initial parameters (like those described [here](https://github.com/NIVANorge/lake_liming_app/issues/11#issuecomment-1438326085)) and run the model "forwards" to simulate changes in Ca concentration over time. Once we can do this, it's relatively easy to specify a "target" change in pH or Ca and then iterate over various combinations of lime products and doses.

Because the Ca time series is so fundamental, I think it makes sense to start by simulating this over time, for a well-defined set of input parameters. For example, I only recently discovered the additional equations required for depth correction, changing hydrology, wet vs. dry spreading etc. These processes only came up in the context of discussing how to simulate the Ca time series (and, based on recent discussions, it seems there may be other issues too). Once we're happy with the structure of the model, we can consider "inverting" it for e.g. lime optimisation, if that's what's desired. 

> In the same plot there's different residency times, does that mean that this part of our scope has changed? "(...) the application will by default assume wet spreading of lime by boat to a lake with residence time 0.7 years. "

Those are still the defaults (i.e. the app should be initialized with `spr_meth = 'wet'` and `res_time = 0.7`), but it was always the intention to make these parameters user-adjustable. We don't necessarily need to produce facet plots like the ones in my notebooks, though - those are for testing to convince myself that the curves behave sensibly as the input parameters are varied.

There are other aspects of the project where I'm trying to resist "scope creep", as there have been some suggestions that the old TPKALK model requires further _scientific_ development to make it suitable for modern conditions. Developing an entirely new model is beyond scope for this small project, though, so it's something I need to discuss further with MDir.
